packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -29,12 +31,29 @@ const _foregroundNotificationEventChannel = EventChannel(
   'com.amazonaws.amplify/push_notification/event/FOREGROUND_MESSAGE_RECEIVED',
 );
 
-const _externalHandleKey = 'externalHandleKey';
+/// {@template amplify_push_notifications.external_handle_key}
+/// Key used to store the external callback handle function.
+///
+/// Used in this file to store the callback function and used in Flutter API class to access it.
+/// {@endtemplate}
+const externalHandleKey = 'externalHandleKey';
 
 const _needsConfigurationException = PushNotificationException(
   'Configure Amplify with Notifications Plugin before using this method.',
 );
 
+/// {@template amplify_push_notifications.android_check_override}
+/// Internal Platform check exposed for testing purposes only.
+/// {@endtemplate}
+@visibleForTesting
+const androidCheckOverride = #_androidCheckOverride;
```

Comment:
This basically reimplements the functionality of `package:os_detect`. I would probably recommend moving to that. You can see example usage in `aws_common`.